samples/common/scripts/generate-keys.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import subprocess
+
+import asyncclick as click
+from golioth import Client
+
+ROOT_CA_NAME = 'root-ca'
+
+def run(*args, **kwargs):
+    if 'check' not in kwargs:
+        kwargs['check'] = True
+
+    if 'shell' not in kwargs:
+        kwargs['shell'] = True
+
+    return subprocess.run(*args, **kwargs)
+
+@click.group()
+async def cli():
+    pass
+
+@click.command()
+@click.option('--upload/--no-upload', default=False, help='Upload root certificate to Golioth')
+@click.option('-c', '--config-path', type=Path,
+              help='Path to goliothctl configuration')
+@click.option('--api-key', help='API key')
+@click.option('--project', 'project_name', help='Project where root certificate will be uploaded')
+@click.option('-o', '--output', type=Path, default=Path('.'), help="Output directory")
+async def root(upload, config_path, api_key, project_name, output):
+    if upload:
+        if not api_key:
+            click.echo("API key required to upload root certificate")
+
+        client = Client(config_path, api_key=api_key)
+
+        if project_name:
+            project = await client.project_by_name(project_name)
+        else:
+            project = await client.default_project()
+
+    root_ca_crt = output / f'{ROOT_CA_NAME}-{project_name}.crt.pem'
+    root_ca_key = output / f'{ROOT_CA_NAME}-{project_name}.key.pem'
+
+    # Generate an elliptic curve private key
+    # Run `openssl ecparam -list_curves` to list all available algorithms
+    # Keep this key safe! Anyone who has it can sign authentic-looking device certificates
+    run(f'openssl ecparam -name prime256v1 -genkey -noout -out "{root_ca_key}"')
+
+    # Create and self-sign a corresponding public key / certificate
+    run(f'openssl req -x509 -new -nodes -key "{root_ca_key}" -sha256 -subj "/C=US/CN=Root {ROOT_CA_NAME}" -days 1024 -out "{root_ca_crt}"')
+
+    if upload:
+        # Add (upload) generated CA certificate to Golioth
+        with root_ca_crt.open('rb') as fp:
+            await project.certificates.add(fp.read(), cert_type='root')
+
+@click.command()
+@click.option('--device-id', help="Primary Device Hardware ID (e.g. a serial number)",
+              required=True)
```

Comment:
In this line there are 3 things mentioned (device id, primary hardware id, serial number) and it is hard to figure out is really requested from the user. Also `(e.g. a serial number)` suggests that this is **something** that user can select (which to some extent is true, but probably not at this point anymore). I would prefer to keep using "primary hardware id" or "hardware id".
```suggestion
@click.option('--primary-hardware-id', help="Primary Device Hardware ID,
              required=True)
```

Reply:
So what this actually will map to in the console is the "Device Name". This goes into the `CN` field in the certificate, which directly populates the Device Name field in the console. The device name is combined with a timestamp to create the "Primary Hardware ID" in the console. It definitely is something that a user can select, and I think a serial number is the best name for most people. I'll update the naming and help text appropriately.

As an aside, I think we should change the way we provision devices to omit the timestamp - I'm going to open a ticket for that with the cloud team.

Reply:
When I list devices with `goliothctl` I get the following:
```
id:"6033cc457016b281d671df53"  hardware_ids:"DE:AD:BE:EF"  name:"nrf52"
...
id:"63989e55811d6b602efec176"  hardware_ids:"20221213154629-de-ad-be-ef"  name:"DE:AD:BE:EF"
```
All my comments about were with the assumption that I was using device `hardware_ids:"DE:AD:BE:EF"  name:"nrf52"`... which seems that I was wrong?

https://www.notion.so/golioth/Device-Certificate-Creation-Guide-f6d859731c784128b7de33d152e11bdd#cde455fa44dd40b185b186ff1dd8f5a9 was used to develop cert-based auth. I am pretty sure that when asking verbally about what the "primary hardware id" means I got a response something along the line "it is the first hardware id".

So... I used `DE:AD:BE:EF` to generate certificates. Device connected. I probably used a `goliothctl` command that shows all the logs (since it was the only service verified after cert-based auth) from all devices. It looked like "my" (`name:"nrf52"`) device has connected. What happened instead is that new device was created and generated hardware-id: `hardware_ids:"20221213154629-de-ad-be-ef"  name:"DE:AD:BE:EF"`. I am surprised, just learned about it seconds ago.

This problem is not related now to this PR. We need to fix our documentation and do not cause any confusion. Question is: what was the intended behavior? Should the "name" be used or "hardware id"?

Reply:
Yeah, I think the CN should map to directly to the "primary hardware id", and also populate the name (just because the device needs a name when it gets created). And then users can go in and change the name in the console if they like.

That actually brings up another problem: that you can change both device name and hardware id on the console - which would break the association between a physical device and its entry in the console.

I agree, fixing this is outside of the scope of this PR.